internal/primitives/state-machine/overlayed_entry.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2025 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package statemachine
+
+import "math"
+
+const PROOF_OVERLAY_NON_EMPTY = `
```

Comment:
do we need to export this?
```suggestion
const proofOverlayNonEmpty = `
```